src/OpenSpaceToolkit/Astrodynamics/Trajectory/Orbit/Models/Kepler/BrouwerLyddaneMeanLong.cpp

```diff
@@ -0,0 +1,436 @@
+using ostk::core::types::Integer;
+
+using EarthGravitationalModel = ostk::physics::environment::gravitational::Earth;
+
+BrouwerLyddaneMeanLong::BrouwerLyddaneMeanLong(
+    const Length &aSemiMajorAxis,
+    const Real &anEccentricity,
+    const Angle &anInclination,
+    const Angle &aRaan,
+    const Angle &anAop,
+    const Angle &aMeanAnomaly
+)
+    : BrouwerLyddaneMean(aSemiMajorAxis, anEccentricity, anInclination, aRaan, anAop, aMeanAnomaly)
+{
+}
+
+COE BrouwerLyddaneMeanLong::toCOE() const
+{
+    // ref:
+    // https://github.com/ChristopherRabotin/GMAT/blob/37201a6290e7f7b941bc98ee973a527a5857104b/src/base/util/StateConversionUtil.cpp#L4206
+
+    const auto mod = [](const Real &value, const Real &bound) -> Real
+    {
+        return value - bound * std::floor(value / bound);
+    };
+
+    // TBM: Make configurable?
+    const Real equatorialRadius = EarthGravitationalModel::EGM2008.equatorialRadius_.inMeters();
+
+    Integer pseudoState = 0;
+    bool possibleInaccuracyLongWritten = false;
+    bool criticalInclinationWritten = false;
+
+    const Real re = EarthGravitationalModel::EGM2008.equatorialRadius_.inMeters();
+    const Real j2 = 1.082626925638815E-03;
+    const Real j3 = -0.2532307818191774E-5;
+    const Real j4 = -0.1620429990000000E-5;
+    const Real j5 = -0.2270711043920343E-6;
+    const Real ae = 1.0;
+    Real smadp = semiMajorAxis_.inMeters() / re;
+    Real eccdp = eccentricity_;
+    Real incdp = inclination_.inRadians();
+    Real raandp = raan_.inRadians(0.0, Real::TwoPi());
+    Real aopdp = aop_.inRadians(0.0, Real::TwoPi());
+    Real meanAnom = anomaly_.inRadians(0.0, Real::TwoPi());
+
+    if (incdp > 3.0543261909900763)
+    {
+        incdp = Real::Pi() - incdp;
+        raandp = -raandp;
+        pseudoState = 1;
+    }
+
+    // negative eccentricity aviodance lines
+    if (eccdp > 0.99)
+    {
+        throw ostk::core::error::RuntimeError(
+            String::Format("BrouwerLyddaneMeanLong is applicable only if 0.0 < Ecc: [{}] < 0.99.", eccdp)
+        );
+    }
+
+    const Real perigee = this->getPeriapsisRadius().inMeters();
+    if (perigee < 3000000.0)
+    {
+        throw ostk::core::error::RuntimeError(
+            String::Format("BrouwerLyddaneMeanLong is applicable only if Perigee radius: [{}] > 3000000.0 m.", perigee)
+        );
+    }
+    if (perigee < 6378000.0)
+    {
+        if (!possibleInaccuracyLongWritten)
+        {
+            std::cout << "Warning: For BrouwerLyddaneMeanLong, when RadPer < 6378km, there "
+                         "is a possible inaccuracy due to singularity related "
+                         "with inside-of-earth orbit.\n";
+            possibleInaccuracyLongWritten = true;
+        }
+    }
+
+    if (incdp > Real::Pi())
+    {
+        throw ostk::core::error::RuntimeError(String::Format(
+            "BrouwerLyddaneMeanLong is applicable only if inclination: [{}] < 180.0 degrees.",
+            Angle::Radians(incdp).inDegrees()
+        ));
```

Comment:
```suggestion
        throw ostk::core::error::RuntimeError(
            "BrouwerLyddaneMeanLong is applicable only if inclination: [{}] < 180.0 degrees.",
            Angle::Radians(incdp).inDegrees()
        );
```
nit: This should work. Runtime error has auto formatting